ui/src/app/domains/views/DomainDetails/ResourceRecords/EditRecordDomainForm.tsx

```diff
@@ -0,0 +1,137 @@
+
+type Props = {
+  id: Domain["id"];
+  resource: DomainResource;
+  closeForm: () => void;
+};
+
+export type CreateRecordValues = {
+  name: DomainResource["name"];
+  rrtype: DomainResource["rrtype"];
+  rrdata: DomainResource["rrdata"];
+  ttl: DomainResource["ttl"] | "";
+};
+
+const EditRecordDomainForm = ({
+  id,
+  resource,
+  closeForm,
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const errors = useSelector(domainSelectors.errors);
+  const saved = useSelector(domainSelectors.saved);
+  const saving = useSelector(domainSelectors.saving);
+  const cleanup = () => dispatch(domainActions.cleanup());
+
+  const CreateRecordSchema: SchemaOf<CreateRecordValues> = Yup.object()
+    .shape({
+      name: Yup.string().required("This field is required name"),
+      rrtype: Yup.string().required("This field is required tuype"),
+      rrdata: Yup.string().required("This field is required data"),
+      ttl: Yup.number()
+        .nullable(true)
+        .min(0, "Ensure this value is greater than or equal to 0."),
+    })
+    .defined();
+
+  const updateAddressRecord = (values: CreateRecordValues) => {
+    const params: UpdateAddressRecordParams = {
+      ...resource,
+      domain: id,
+      previous_name: resource["name"],
+      previous_rrdata: resource["rrdata"],
+      previous_rrtype: resource["rrtype"],
+      previous_ttl: resource["ttl"],
+      name: values.name,
+      rrdata: values.rrdata,
+      rrtype: values.rrtype,
+      address_ttl:
+        values.ttl === "" || values.ttl === null ? null : Number(values.ttl),
+      ip_addresses: (values.rrdata ?? "").split(/[ ,]+/),
+    };
+
+    if (values.name !== resource.name) {
+      dispatch(domainActions.updateDnsResource(params));
+    }
+
+    dispatch(domainActions.updateAddressRecord(params));
+  };
+
+  const updateDnsRecord = (values: CreateRecordValues) => {
+    const params: UpdateResourceParams = {
+      ...resource,
+      domain: id,
+      previous_name: resource["name"],
+      previous_rrdata: resource["rrdata"],
+      previous_rrtype: resource["rrtype"],
+      previous_ttl: resource["ttl"],
+      name: values.name,
+      rrdata: values.rrdata,
+      rrtype: values.rrtype,
+      ttl: values.ttl === "" || values.ttl === null ? null : Number(values.ttl),
+    };
+
+    if (values.name !== resource.name) {
+      dispatch(domainActions.updateDnsResource(params));
+    }
+
+    dispatch(domainActions.updateDnsData(params));
+  };
+
+  const updateRecord = (values: CreateRecordValues) => {
+    cleanup();
+
+    if (values.rrtype === RecordType.A || values.rrtype === RecordType.AAAA) {
+      updateAddressRecord(values);
+    } else {
+      updateDnsRecord(values);
+    }
+  };
+
+  return (
+    <FormikForm<CreateRecordValues>
+      buttonsBordered={false}
+      // FIXME
+      // this form seems to be unmounted after submit, so errors are cleared before they can render
+      // cleanup={cleanup}
```

Comment:
@Caleb-Ellis @huwshimi I wasn't able to debug what happens there, but it seemed that when I passed the cleanup into the FormikForm the errors were cleared too often.

When form was submitted the errors didn't even show up, because form was unmounted and rendered again, causing cleanup to be called, and cleared the errors. 
I'm not sure why the form is being unmounted after submit. I was trying to debug if it happens because of expandable table or something, but didn't find the reason.

So here cleanup is commented out of formik form, but it is called manually in other places (like when table is expanded), so we control when errors are cleared.